src/pycirclizely/utils/hover_formatter.py

```diff
@@ -0,0 +1,212 @@
+
+        # Handle colors - accept either single color or list
+        if isinstance(colors, str):
+            self.hover_colors = [colors] * len(x)
+        else:
+            if len(colors) != len(x):
+                raise ValueError(
+                    f"colors length ({len(colors)}) must match data length ({len(x)})"
+                )
+            self.hover_colors = colors
+
+    def process_stacked_bar_hover_text(
+        self,
+        sb_table: StackedBarTable,
+        col_name2color: Dict[str, str],
+        hover_text: HoverText = "default",
+    ) -> None:
+        """
+        Process hover text for stacked bar plots.
+
+        Args:
+            hover_text: Either "default", a list of strings, or None
+            sb_table: StackedBarTable object
+            col_name2color: Dictionary mapping column names to colors
+        """
+        if hover_text is None:
+            return
+
+        if hover_text == "default":
+            self.hover_texts = self._default_stackedbar_hovertext(sb_table)
+        elif isinstance(hover_text, list):
+            expected_length = len(sb_table.row_names) * len(sb_table.col_names)
+            if len(hover_text) != expected_length:
+                raise ValueError(
+                    f"""hover_text length ({len(hover_text)}) must match number of
+                     segments ({expected_length})"""
+                )
+            self.hover_texts = hover_text
+        else:
+            raise TypeError("hover_text must be 'default', list[str], or None")
+
+        self.hover_colors = []
+        for col_name in sb_table.col_names:
+            color = col_name2color[col_name]
+            self.hover_colors.extend([color] * len(sb_table.row_names))
+
+    def add_hover_positions(
+        self,
+        rad: NumericSequence,
+        r: NumericSequence,
+    ) -> None:
+        """
+        Add hover positions from polar coordinates.
+
+        Args:
+            rad: Angular coordinates in radians
+            r: Radial coordinates
+        """
+        for theta, rho in zip(rad, r):
+            cx, cy = PolarSVGPatchBuilder._polar_to_cart(theta, rho)
+            self.hover_x.append(cx)
+            self.hover_y.append(cy)
+
+    def create_hover_trace(self, **kwargs) -> BaseTraceType:
+        """Create a Plotly dummy hover trace."""
+        args = dict(
+            text=self.hover_texts,
+            marker=dict(color=self.hover_colors),
+            hoverlabel=dict(bgcolor=self.hover_colors),
+        )
+
+        # Start from defaults, then update with args, then with kwargs
+        all_args = deep_dict_update(
+            deep_dict_update(config.hovertext_dummy_defaults, args), kwargs
+        )
+
+        return build_scatter_trace(
+            x=self.hover_x, y=self.hover_y, mode="markers", **all_args
+        )
+
+    def clear(self) -> None:
+        """Clear all stored hover data."""
+        self.hover_texts = []
```

Comment:
I've seen cases where an external class gets a reference to a list field, and then after clear you end up with duplicated list instances.
To avoid this, do self.hover_texts.clear()